### PR TITLE
chore: alert banner for HashiConf

### PR DESCRIPTION
**Pull request by heatlikeheatwave**
## 🔗 Relevant links

- [Preview](https://dev-portal-git-heat-hashiconfalert-banner-day-one-hashicorp.vercel.app/)
- [Asana task](https://app.asana.com/0/1207947026228781/1208503549555255/f) 🎟️

## 🗒️ What

- alert banner to watch live stream of HashiConf on October 15th, 2024 8:30 am - 12:30 pm EDT

## 🧪 Testing

- Visit the [preview link ](https://dev-portal-git-heat-hashiconfalert-banner-day-one-hashicorp.vercel.app/)and see the alert banner is not there
- Change the date & time on your machine to October 15th 8:30 AM and refresh the page
- The alert banner should be visible
- Change the date & time on your machine to October 15th 12:31 PM and refresh the page
- The alert banner should **not** be visible

### CODE REVIEW

File: src/layouts/base-layout/index.tsx [hunk 1]
```diff
@@ -60,11 +60,42 @@ const BaseLayout = ({
 	useScrollPercentageAnalytics()
 	const [showSkipLink, setShowSkipLink] = useState(false)
 
+	// TODO: remove after HashiConf 2024
+	// start of code to remove
+	const [isDuringHashiConf, setIsDuringHashiConf] = useState(false)
+
+	useEffect(() => {
+		const now = new Date().getTime() // current date & time
```
Comment (RubenSandwich) [inline src/layouts/base-layout/index.tsx:68]: @heatlikeheatwave we have a timezone bug here. Namely we are getting a random timezone but then comparing it to PST, which could mean they are already in Hashiconf time when Hashiconf hasn't even started yet. We need to get the users timezone in PST to accurately compare to PST.

So now should be:
`const now = new Date().toLocaleString('en-US', { timeZone: 'America/Los_Angeles' })`

**Review by RubenSandwich — CHANGES_REQUESTED** [review]
We have a timezone bug as we are compare across timezones.

File: src/layouts/base-layout/index.tsx (hunk 1 above)
Comment (RubenSandwich) [reply src/layouts/base-layout/index.tsx:68]: Actually that is just going to just return the string. I think we should import something like: https://day.js.org/en/ and use it's date manipulation functions as that is going to make comparing across timezones easier.

Comment (heatlikeheatwave) [reply src/layouts/base-layout/index.tsx:68]: Shouldn't it be EDT since the conference is in Boston?

Comment (heatlikeheatwave) [reply src/layouts/base-layout/index.tsx:68]: Also, great catch with this!

Comment (RubenSandwich) [reply src/layouts/base-layout/index.tsx:68]: Dates. Why you so hard. 😢

Comment (aaronvanderlip) [reply src/layouts/base-layout/index.tsx:68]: Y'all are making it happen with more 👁️ on the code.

Comment (heatlikeheatwave) [reply src/layouts/base-layout/index.tsx:68]: @RubenSandwich I tested this again, and I don't think this has the bug you described. I changed my location and time on my machine to west coast and europe; the visibility of the banner worked as expected. What am I missing that you're seeing? 

I do agree that the pacific time zone (.e.g -8:00 in the time stamp) can be changed. That was a mistake on my part.

File: src/layouts/base-layout/index.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@
 // Third-party imports
 import classNames from 'classnames'
 import AlertBanner from '@hashicorp/react-alert-banner'
-import { HTMLAttributes, useState } from 'react'
+import { HTMLAttributes, useEffect, useState } from 'react'
```
Comment (aaronvanderlip) [inline src/layouts/base-layout/index.tsx:9]: I don't think the `useEffect` is used? But as long as the linter doesn't complain and it builds I have no issue with it.

**Review by aaronvanderlip — APPROVED** [review]

**Pull request merged**